tests/unittests/test_dispatcher.py

```diff
@@ -392,4 +352,39 @@ async def _check_if_async_function_is_ok(self, host) -> Tuple[str, str]:
         self.assertEqual(call_r.response.result.status,
                          protos.StatusResult.Success)
 
-        return (func_id, invoke_id)
+        return func_id, invoke_id
+
+
+class TestThreadPoolSettingsPython38(TestThreadPoolSettingsPython37):
+    def setUp(self):
+        super(TestThreadPoolSettingsPython38, self).setUp()
+        self.mock_version_info = patch(
+            'azure_functions_worker.dispatcher.sys.version_info',
+            SysVersionInfo(3, 8, 0, 'final', 0))
+        self.mock_version_info.start()
+
+    def tearDown(self):
+        os.environ.clear()
+        os.environ.update(self._pre_env)
+        self.mock_version_info.stop()
+
+
+class TestThreadPoolSettingsPython39(TestThreadPoolSettingsPython37):
+    def setUp(self):
+        super(TestThreadPoolSettingsPython39, self).setUp()
+
+        # 6 - based on 2 cores - min(32, (os.cpu_count() or 1) + 4) - 2 + 4
+        self._default_workers: Optional[int] = 6
+        self.mock_os_cpu = patch(
```

Comment:
We also want to start the patching mock_os_cpu here

Reply:
Ah.. :) I missed this and this failed.

You should never do changes at 2 AM. :)